analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -302,3 +330,25 @@ internal fun Identifier.toVcsInfo(): VcsInfo {
     val vcsInfo = VcsHost.toVcsInfo("https://$name")
     return vcsInfo.copy(revision = getRevision(version))
 }
+
+/**
+ * Parse the given [lines] generated by the _go mod why_ command and the names of packages that are reported to be
+ * used by the main module.
+ * See https://golang.org/ref/mod#go-mod-why.
+ */
+internal fun parseWhyOutput(lines: Sequence<String>): Set<String> {
+    val usedPackages = mutableSetOf<String>()
+    var currentPackage: String? = null
+
+    lines.forEach { line ->
+        if (line.startsWith(PACKAGE_SEPARATOR)) {
+            currentPackage = line.substring(PACKAGE_SEPARATOR.length)
```

Comment:
consider using `.removePrefix`

Reply:
... or `.withoutPrefix`.

Reply:
I went for `removePrefix()`, although I think the implementation with `substring()` is probably slightly more efficient, as the `startsWith()` check has to be done anyway.

Reply:
That was exactly the movitation to introduce `withoutPrefix()` which allows you to do:

    lines.forEach { line ->
        line.withoutPrefix(PACKAGE_SEPARATOR)?.let {
            currentPackage = it
        } ?: run {
            if (!line.startsWith('(') && line.isNotBlank()) {
                currentPackage?.let { usedPackages += it }
            }
        }
    }

Reply:
Indeed stylish. But for the problem at hand, I find the current code more readable; it better expresses that the line is checked for different patterns which trigger specific actions.